### PR TITLE
Feat: improve transpilation of CREATE TABLE LIKE statement

### LINKED ISSUE #2922: Transpiling a CREATE TABLE ... LIKE ... statement to DuckDB does not use the existing DuckDB syntax
**Fully reproducible code snippet**
```python
import sqlglot

sqlglot.transpile("CREATE TABLE t2 LIKE t1", read="snowflake", write="duckdb")
```

**Official Documentation**
https://duckdb.org/docs/sql/statements/create_table.html

While DuckDB does not support the LIKE property, but the same can be expressed with:
```SQL
CREATE TABLE t2 AS FROM t1 LIMIT 0;
```

### PR DESCRIPTION
**Pull request by georgesittas**
Fixes #2922

We can probably transfer this logic to the base generator and control it with a flag, so other dialects besides duckdb can leverage it. Can open a separate PR for it later or do it here, just need a bit more time to investigate.

### CODE REVIEW

**Review by tobymao — APPROVED** [review]
a flag would be nice

**Comment by georgesittas** [comment]
yep doing that work now

**Comment by georgesittas** [comment]
Improved the transpilation as discussed above.

References:
- https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_table_like
- https://clickhouse.com/docs/en/sql-reference/statements/create/table
- https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-like.html
- https://doris.apache.org/docs/dev/sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-TABLE-LIKE/
- https://dev.mysql.com/doc/refman/8.0/en/create-table-like.html
- https://docs.oracle.com/cd/E17952_01/mysql-8.0-en/create-table-like.html
- https://www.postgresql.org/docs/current/sql-createtable.html
- http://prestodb.io/docs/0.285/sql/create-table.html
- https://trino.io/docs/current/sql/create-table.html
- https://stackoverflow.com/questions/616104/what-is-the-equivalent-of-create-table-like-in-sql-server
- https://www.sqlite.org/lang_createtable.html

File: sqlglot/generator.py [hunk 1]
```diff
@@ -1274,9 +1280,21 @@ def property_sql(self, expression: exp.Property) -> str:
...
+            like = f"LIKE {self.sql(expression, 'this')}{options}"
+            if self.LIKE_PROPERTY_INSIDE_SCHEMA and not isinstance(expression.parent, exp.Schema):
+                like = f"({like})"
+
+            return like
+
+        if expression.expressions:
+            self.unsupported("Transpilation of LIKE property options is unsupported")
+
+        select = exp.select("*").from_(self.sql(expression, "this")).limit(0)
```
Comment (tobymao) [inline sqlglot/generator.py:1296]: why do you call sql on the from_? can you just pass it in?

**Review by tobymao — APPROVED** [review]

File: sqlglot/generator.py (hunk 1 above)
Comment (georgesittas) [reply sqlglot/generator.py:1296]: ah missed it, thanks

**Pull request merged**